transaction/std/src/transaction_builder.rs

```diff
@@ -427,7 +427,8 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
             create_output_with_fog_hint(self.block_version, amount, recipient, hint, memo_fn, rng)?;
 
         let (amount, blinding) = tx_out
-            .masked_amount
+            .get_masked_amount()
+            .expect("TransactionBuilder created an invalid VersionedMaskedAmount")
```

Comment:
:wrench: This was here before with the expect 2 lines below, but it seems inconsistent that this error will cause the program to abort, while other errors in this function keep operating.
Perhaps since there is now ` TxOutConversion(TxOutConversionError),` these might return the error.

Reply:
my thinking here was that this error should never occur in prod, it's a logic error if this operation fails and i'm basically making an assertion that it doesn't fail.

I'm okay to propagate it like you say, I think it doesn't make that much difference

Reply:
changing this in a nice way means changing a bunch of surrounding code in a way that's orthogonal to this PR, so I'd rather do this in a later PR. Let's make a ticket if you want to track this

Reply:
I may have latched on to the _different_ error cases too much and not scrutinized the `expect` statement
```rust
.expect("TransactionBuilder created an invalid MaskedAmount")
```
That implies the `TransactionBuilder` failed, thus an internal code error.  Maybe there is better wording, but I think the fact that it's referencing internal types should communicate an internal implementation failure, versus an external communication or malformed data error.

My initial thought was to add a `// SAFETY: ` comment like was done in the 512bit hash -> `[u8; 32]`.  But I think when used diligently, the `expect("reasoning")` should fulfill that more concisely and provide better error handling should it happen.